trustmanager/keystore.go

```diff
@@ -142,13 +142,12 @@ func (s *GenericKeyStore) AddKey(keyInfo KeyInfo, privKey data.PrivateKey) error
 	name := filepath.Join(keyInfo.Gun, privKey.ID())
 	for attempts := 0; ; attempts++ {
 		chosenPassphrase, giveup, err = s.PassRetriever(name, keyInfo.Role, true, attempts)
-		if err != nil {
-			continue
+		if err == nil {
+			break
 		}
 		if giveup || attempts > 10 {
```

Comment:
Non-blocking:  wondering if we should decrease this number to 3 or some such?  That would give the user 5 tries.  (currently it fails out after 12 tries - I was wondering whether that's too many)